lib/howl/buffer.moon

```diff
@@ -277,6 +278,19 @@ class Buffer extends PropertyObject
       b_end_pos = @byte_offset end_pos
       @_buffer\ensure_styled_to pos: b_end_pos
 
+  mode_at: (pos) =>
+    b_pos = @byte_offset pos
+    mode_name = @_buffer.styling\get_mode_name_at b_pos
+    -- Returns @mode if there's no marker or the requested mode doesn't exist.
+    mode_name and mode.by_name(mode_name) or @mode
+
+  config_at: (pos) =>
+    mode_at = @mode_at pos
+    return @config if mode_at == @mode
+    new_config = config.local_proxy!
```

Comment:
@nilnor The only thing it's used for is `word_pattern`. My proposed fix was to just change `word_pattern` to be a buffer setting, not a config option, since I can't think of a single use case for actually changing that.